posts/2021-05-07-caught-red-handed.md

```diff
@@ -0,0 +1,125 @@
+[profiles]: https://doc.rust-lang.org/cargo/reference/profiles.html
+
+If your project has not opted into enabling incremental compilation, then none of the fingerprint issues should affect your release builds.
+
+## What should a Rust programmer do in response
+
+The Internal Compiler Error asks you to report a bug, and if you can do so, we still want that information. We *want* to know about the cases that are failing.
+
+But regardless of whether or not you file a bug, the problem here can be resolved by either:
+
+ 1. deleting your incremental compilation cache (e.g. by running `cargo clean`), or
+ 2. force incremental compilation to be disabled, by setting `CARGO_INCREMENTAL=0` in your environment or `build.incremental` to `false` in the `config.toml`.
+
+We recommend that users of 1.52.0 disable incremental compilation, to avoid running into this problem.
+
+We do *not* recommend that users of 1.52.0 downgrade to an earlier version of Rust in response to this problem. As noted above, there is at least one instance of a silent [miscompilation][issue-82920] caused by incremental compilation that was not caught until we added the fingerprint checking.
+
+## What is the Rust project going to do to fix this
+
+### Short-term plan
+
+Based on the number of bug reports that have already come in, we know we need to do something quickly.
+
+We are going to be issuing a point release, 1.52.1. The point release will improve the diagnostic output from `verify-ich`, so that a programmer can more effectively act in response to the message (i.e., they will be told that the problem is due to the use of incremental compilation, and that they need to delete their cache, at the very least).
+
+We might also make incremental compilation opt-in for *all* Cargo [profiles][]; that is, we may turn incremental compilation off for the  `dev` and `test`, so that they match `release` and `bench`, which already have incremental compilation off. (This remains to be decided.)
+
+We do not plan to disable incremental compilation in its entirety. We believe that incremental compilation is providing value for a large number of Rust programmers, and we want to empower them to continue using it, as long as they are equipped with the tools they need to turn it off if they encounter this issue.
+
+### Long-term plan
+
+The long-term plan is to fix the bugs! Incremental compilation is the only realistic way for the Rust compiler to be able to provide a fast edit-compile-run cycle for all of its programmers, and so we need to address [all of the issues][issue-list] that have been identified thus far via `verify-ich`. (There are 32 such issues as of this writing, though many may be duplicates.)
+
+[issue-list]: https://github.com/rust-lang/rust/issues?q=is%3Aissue+is%3Aopen+unstable+fingerprints
+
+If you want to come help us do it, we would love for you to [join us][] with the effort!
+
+[join us]: https://www.rust-lang.org/community
```

Comment:
I think a more specific place for people to go who want to help with this specific problem would be better than rust-lang.org/community-- would the T-compiler stream on zulip be appropriate? WG-incr-comp? Maybe have a transient stream just for this issue so that the more general streams can be about other topics?